lib/chef/knife/cookbook_site_install.rb

```diff
@@ -141,10 +141,16 @@ def download_cookbook_to(download_path)
 
       def extract_cookbook(upstream_file, version)
         ui.info("Uncompressing #{@cookbook_name} version #{version}.")
-        # FIXME: Detect if we have the bad tar from git on Windows: https://github.com/opscode/chef/issues/1753
         extract_command = "tar zxvf \"#{convert_path upstream_file}\""
         if Chef::Platform.windows?
-          extract_command << " --force-local"
+          # FIXED: Detect if we have the bad tar from git on Windows: https://github.com/opscode/chef/issues/1753
+          tar_version = shell_out("tar --version").stdout.tr("\n", " ")
+          if tar_version =~ /GNU tar/
+            ui.info("GNU tar detected")
+            extract_command << " --force-local"
+          elsif tar_version =~ /bsdtar/
+            ui.info("BSD tar detected")
```

Comment:
that should probably be debug level output rather than ui.info.

the elsif is also not terribly useful since its mostly a no-op, and there's the third case which is a missing else, and it doesn't really communicate what its doing when it detects it...

```
if tar_version =~ /GNU tar/
  ui.debug("GNU tar detected, adding --force-local")
  extract_command << " --force-local"
else
  ui.debug("non-GNU tar detected, not adding --force-local")
end
```